src/main/java/com/google/devtools/build/lib/bazel/rules/python/python_stub_template.txt

```diff
@@ -275,6 +275,8 @@ def Main():
   python_path_entries = CreatePythonPathEntries(python_imports, module_space)
   python_path_entries += GetRepositoriesImports(module_space, %import_all%)
 
+  # Remove any duplicates to reduce path length
+  python_path_entries = set(python_path_entries)
```

Comment:
A set will scramble the order of these entries, possibly non-deterministically. To preserve existing behavior (in most cases) let's keep the existing order and filter out all but the first occurrence of each entry.